Makefile

```diff
@@ -1,14 +1,26 @@
 PROJECT_NAME=qdrouterd
 DOCKER_REGISTRY=quay.io
 PWD=$(shell pwd)
-ROUTER_SOURCE_URL=http://archive.apache.org/dist/qpid/dispatch/1.8.0/qpid-dispatch-1.8.0.tar.gz
-PROTON_SOURCE_URL=http://archive.apache.org/dist/qpid/proton/0.28.0/qpid-proton-0.28.0.tar.gz
-
+ # This is the latest version of the Qpid Dispatch Router
+DISPATCH_VERSION=1.8.0
+PROTON_VERSION=0.28.0
+PROTON_SOURCE_URL=http://archive.apache.org/dist/qpid/proton/${PROTON_VERSION}/qpid-proton-${PROTON_VERSION}.tar.gz
+# Look for environment variable DOCKER_TAG. If available use it in the URL for obtaining the router source code
+# and use it as the docker tag when pushing the image.
+# If DOCKER_TAG is not available, push the image with the "latest" tag using the router version DISPATCH_VERSION
+ifneq ($(strip $(DOCKER_TAG)),)
+	DOCKER_TAG_VAL=$(DOCKER_TAG)
+	ROUTER_SOURCE_URL=http://archive.apache.org/dist/qpid/dispatch/${DOCKER_TAG_VAL}/qpid-dispatch-${DOCKER_TAG_VAL}.tar.gz
+else
+	DOCKER_TAG_VAL=latest
+	ROUTER_SOURCE_URL=http://archive.apache.org/dist/qpid/dispatch/${DISPATCH_VERSION}/qpid-dispatch-${DISPATCH_VERSION}.tar.gz
```

Comment:
I think it would be more intuitive to always use DISPATCH_VERSION for the source url. If DOCKER_TAG is specified that would be used as the tag to be pushed, if not it would default to the DISPATCH_VERSION. And if LATEST_TAG was set it would additionally tag it as latest perhaps and push that? (Thanks again for doing this, it is looking good!)

Reply:
So let's see
1. DISPATCH_VERSION will be a local variable in the Makefile and will be hardcoded/defaulted to 1.8.0. It can be changed to 1.9.0 when a new version comes out and so on. DISPATCH_VERSION will always be used to construct the ROUTER_SOURCE_URL like this - 
ROUTER_SOURCE_URL=http://archive.apache.org/dist/qpid/dispatch/${DISPATCH_VERSION}/qpid-dispatch-${DISPATCH_VERSION}.tar.gz

2. If DOCKER_TAG Makefile parameter is specified (e.g. make push DOCKER_TAG=1.6.0) that would be used as the tag to be pushed, if not it would default to the DISPATCH_VERSION.
Quick question here - What if DISPATCH_VERSION is set to 1.8.0 and I say - make; make push DOCKER_TAG=1.6.0? That would download the 1.8.0 version of the router code but push it to quay.io with DOCKER_TAG 1.6.0. Is that good ?


3. if  LATEST_TAG was set it would additionally tag it as latest and push that
How would you pass this to make ?
Like make; make push DOCKER_TAG=1.8.0 LATEST_TAG=latest 
Like that ?

Reply:
> Quick question here - What if DISPATCH_VERSION is set to 1.8.0 and I say - make; make push DOCKER_TAG=1.6.0? That would download the 1.8.0 version of the router code but push it to quay.io with DOCKER_TAG 1.6.0. Is that good ?

No, that would be bad, but I don't think we need the script to prevent the user from doing the wrong thing.

Regarding the LATEST_TAG, now you have written this out, I think maybe we don't need it and the user could just do (after editing to make sure the DISPATCH_VERSION was correct):

```
make
make push
make push DOCKER_TAG=latest
```
I think that would be sufficient, do you agree?

Reply:
And where I say 'user' above, I really mean travis, as this could then all be automated. So process would be to just update the DISPATCH_VERSION (and possibly PROTON_VERSION) and commit, and travis would then build and push as requested.

Reply:
> > Quick question here - What if DISPATCH_VERSION is set to 1.8.0 and I say - make; make push DOCKER_TAG=1.6.0? That would download the 1.8.0 version of the router code but push it to quay.io with DOCKER_TAG 1.6.0. Is that good ?
> 
> No, that would be bad, but I don't think we need the script to prevent the user from doing the wrong thing.
ok, sounds good.
> 
> Regarding the LATEST_TAG, now you have written this out, I think maybe we don't need it and the user could just do (after editing to make sure the DISPATCH_VERSION was correct):
> 
> ```
> make
> make push
> make push DOCKER_TAG=latest
> ```
> 
> I think that would be sufficient, do you agree?
ok, so "make push" would use DISPATCH_VERSION as the tag when pushing to quay.io

Reply:
> And where I say 'user' above, I really mean travis, as this could then all be automated. So process would be to just update the DISPATCH_VERSION (and possibly PROTON_VERSION) and commit, and travis would then build and push as requested.

Yes. agreed.